io.c

```diff
@@ -55,22 +55,22 @@ int connect_ipc_sock(const char *socket_pathname) {
 
   socket_fd = socket(AF_UNIX, SOCK_STREAM, 0);
   if (socket_fd < 0) {
-    LOG_ERR("socket() failed for pathname %s.", socket_pathname);
+    LOG_ERR("socket() failed for pathname %s.\n", socket_pathname);
```

Comment:
This \n should be added in LOG_ERR instead of here, to be consistent with the other logging macros. Same for the other calls.